robot_calibration/src/generate_and_publish_joint_states.cpp

```diff
@@ -1,28 +1,57 @@
 #include "ros/ros.h"
 #include "sensor_msgs/JointState.h"
+#include <stdlib.h>
+#include <time.h>
+#include <cmath>
 
 int main(int argc, char **argv)
 {
   ros::init(argc, argv, "simple_joint_state_pub");
 
-  ros::NodeHandle nodeHandler;
+  ros::NodeHandle nh;
 
-  ros::Publisher joint_states_pub = nodeHandler.advertise<sensor_msgs::JointState>("joint_states", 20);
+  ros::Publisher joint_states_pub = nh.advertise<sensor_msgs::JointState>("joint_states", 20);
 
+  int num_state_changes;
+  nh.param("robot_joint_angles/num_state_changes", num_state_changes, 50);
+ 
+  double state_update_interval_float;
+  nh.param("robot_joint_angles/state_update_interval", state_update_interval_float, 3.0);
```

Comment:
Include some error checking to crash the program when the interval is <= 0

Reply:
Decided to go with option 2, issuing a ROS warning and going to default.

Reply:
Next time say what option 2 is, so people other me looking at this knows what is being referred to.